File: v0.1/configurateModel.py
import numpy as np
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense


def create_model(num_q: int, first: int, second: int, num_s: int):
    model = Sequential([
        Dense(first, activation='relu', input_shape=(num_q, )),  
        Dense(second, activation='relu'),  
        Dense(num_s, activation='sigmoid')  
    ])
    # Compile the model
    model.compile(optimizer='adam',
              loss='binary_crossentropy',
              metrics=['accuracy'])

    return model

model = create_model(10,32,32,17)

model.save('./v0.1/model.h5')

File: ib-cs-ia/nn_js/configurateModel.js
const tf = require('@tensorflow/tfjs')

function create_model(input_shape, first, second, output_shape) {
    const model = tf.sequential();
    model.add(tf.layers.dense({units: first, activation: 'relu', inputShape: [input_shape]}));
    model.add(tf.layers.dense({units: second, activation: 'relu'}));
    model.add(tf.layers.dense({units: output_shape, activation: 'sigmoid'}));

    model.compile({optimizer: 'adam', loss: 'binaryCrossentropy', metrics: ['accuracy']});

    return model;
}

async function saveModel(model, path) {
    await model.save(path);
}

model = create_model(10, 32, 32, 17);
saveModel(model, "file://D:/local_repos/CS_IA_Subject_Recommender/ib-cs-ia/nn_js/model");
